sdk/batch.py

```diff
@@ -51,6 +53,7 @@ class Batch:
     jobs: Dict[int, Job] = field(default_factory=dict)
     jobs_count: int = 0
     jobs_count_per_status: Dict[str, int] = field(default_factory=dict)
+    configuration: Optional[Union[Configuration, Dict]] = None
```

Comment:
Why can this be either a Configuration or a Dict object ? Can't we stick to one of the two types with proper casting on batch deserialization?

Otherwise it might be error prone when manipulating inside methods

Reply:
Configuration doesn't apply to the QPU nor EMU_FREE.
I guess we can make it so that we have a Configuration with no fields for those and have inherited classes for the emulators with fields...

The concept of multiple Configuration objects might become reality once we put different types of emulators online.

Reply:
The thing is that we want part of the config to be free-form because we're not totally sure of what that contains, but we  still wanted some structure to help users with the fields we are sure about

Reply:
I don't expect users to create batch object manually, it's always going to be returned by the `get_batch` or `create_batch` function from the SDK. 

We have full control over the attributes deserialization in the batch object, which is why I find it surprising that we cannot decide for the configuration attribute to always be a `dict` or a `Configuration` instance. If configuration is too restrictive for what the backend can send back let's just use a dict for the configuration attribute here.

In the batch creation function inputs however we can be a bit more lenient and allow both types for better user experience.

Reply:
You made a very good point and I did change it. It now only accepts Configuration.

Reply:
Wonderful thanks :+1: